feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,737 @@
+		processName := result.GetName()
+		utilization := result.GetCpuUtilization()
+
+		if utilization > threshold {
+			t.Errorf("%s %s ERROR: %s process %-40s utilization: %3d%%, exceeding %3d%%", timestamp, deviceName, description, processName, utilization, threshold)
+		} else {
+			t.Logf("%s %s INFO: %s process %-40s utilization: %3d%%", timestamp, deviceName, description, processName, utilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighMemorySpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System Process Memory"
+
+	currentTime := time.Now().Round(time.Second)
+	processes := gnmi.GetAll(t, dut, query)
+	for _, process := range processes {
+		processName := process.GetName()
+
+		if process.MemoryUtilization != nil {
+			utilization := process.GetMemoryUtilization()
+			if utilization > memoryHighUtilization {
+				t.Errorf("%s %s ERROR: %s - Process: %-40s - Utilization: %3d%%, exceeding %3d%%", currentTime, deviceName, description, processName, utilization, memoryHighUtilization)
+			} else {
+				t.Logf("%s %s INFO:  %s - Process: %-40s - Utilization: %3d%%", currentTime, deviceName, description, processName, utilization)
+			}
+		} else {
+			t.Logf("%s %s INFO:  %s - Process: %-40s - Utilization data not available", currentTime, deviceName, description, processName)
+		}
+	}
+}
+
+func TestNoQueueDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	type testCase struct {
+		desc     string
+		path     string
+		counters []*ygnmi.Value[uint64]
+	}
+
+	interfaces := sortedInterfaces(dut.Ports())
+	t.Logf("Interfaces: %s", interfaces)
+	for _, intf := range interfaces {
+		qosInterface := gnmi.OC().Qos().Interface(intf)
+		cases := []testCase{
+			{
+				desc:     "Queue Input Dropped packets",
+				path:     "/qos/interfaces/interface/input/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Input().QueueAny().DroppedPkts().State()),
+			},
+			{
+				desc:     "Queue Output Dropped packets",
+				path:     "/qos/interfaces/interface/output/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Output().QueueAny().DroppedPkts().State()),
+			},
+			{
+				desc:     "Queue input voq-output-interface dropped packets",
+				path:     "/qos/interfaces/interface/input/virtual-output-queues/voq-interface/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Input().VoqInterfaceAny().QueueAny().DroppedPkts().State()),
+			},
+		}
+
+		for _, c := range cases {
+			t.Run(c.desc, func(t *testing.T) {
+				if len(c.counters) == 0 {
+					t.Skipf("%s Interface %s Telemetry Value is not present", c.desc, intf)
+				}
+				for queueID, dropPkt := range c.counters {
+					dropCount, present := dropPkt.Val()
+					if !present {
+						t.Errorf("%s Interface %s %s Telemetry Value is not present", c.desc, intf, dropPkt.Path)
+					} else {
```

Comment:
Thanks for your comments. Because the testing only check path present and doesn't compare value, updated the code as below:

```go
					dropCount, present := dropPkt.Val()
					if !present {
						t.Errorf("%s Interface %s %s Telemetry Value is not present", c.desc, intf, dropPkt.Path)
					} else {
						t.Logf("%s Interface %s, Queue %d has %d drop(s)", dropPkt.Path.GetOrigin(), intf, queueID, dropCount)
					}
```